src/data/information_panel.rs

```diff
@@ -0,0 +1,390 @@
+#![cfg(feature = "information_view")]
+
+use crate::{DirectoryEntry, FileDialog};
+use chrono::{DateTime, Local};
+use egui::ahash::{HashMap, HashMapExt};
+use egui::Ui;
+use indexmap::IndexMap;
+use std::fs::File;
+use std::io::{self, Read};
+use std::path::PathBuf;
+
+type SupportedPreviewFilesMap = HashMap<String, Box<dyn FnMut(&mut Ui, &InfoPanelEntry)>>;
+type SupportedAdditionalMetaFilesMap =
+    HashMap<String, Box<dyn FnMut(&mut IndexMap<String, String>, &PathBuf)>>;
+
+fn format_pixels(pixels: u32) -> String {
+    const K: u32 = 1_000;
+    const M: u32 = K * 1_000;
+
+    if pixels >= K {
+        format!("{:.2} MPx", f64::from(pixels) / f64::from(M))
+    } else {
+        format!("{pixels} Px")
+    }
+}
+
+/// Wrapper for the `DirectoryEntry` struct, that also adds the option to store text content
+pub struct InfoPanelEntry {
+    /// Directory Item containing info like path
+    pub directory_entry: DirectoryEntry,
+    /// Optional text content of the file
+    pub content: Option<String>,
+}
+
+impl InfoPanelEntry {
+    /// create a new`InfoPanelEntry` object
+    pub const fn new(item: DirectoryEntry) -> Self {
+        Self {
+            directory_entry: item,
+            content: None,
+        }
+    }
+}
+
+impl InfoPanelEntry {
+    /// Returns the content of the directory item, if available
+    pub fn content(&self) -> Option<&str> {
+        self.content.as_deref()
+    }
+
+    /// Mutably borrow content
+    pub fn content_mut(&mut self) -> &mut Option<String> {
+        &mut self.content
+    }
+}
+
+/// The `InformationPanel` struct provides a panel to display metadata and previews of files.
+/// It supports text-based file previews, image previews, and displays file metadata.
+///
+/// # Fields
+/// - `load_text_content`: Flag to control whether text content should be loaded for preview.
+/// - `supported_files`: A hashmap mapping file extensions to their respective preview rendering functions.
+pub struct InformationPanel {
+    panel_entry: Option<InfoPanelEntry>,
+    /// Flag to control whether text content should be loaded for preview.
+    pub load_text_content: bool,
+    /// Max chars that should be loaded for preview of text files.
+    pub text_content_max_chars: usize,
+    loaded_file_name: PathBuf,
+    supported_preview_files: SupportedPreviewFilesMap,
+    additional_meta_files: SupportedAdditionalMetaFilesMap,
+    other_meta_data: IndexMap<String, String>,
```

Comment:
Could you also write a short description about these variables? In particular, the difference between `supported_preview_files` and `additional_meta_files`.